### PR TITLE
Added provider_job_id to jobba records

### PR DESCRIPTION
**Pull request by Dantemss**
Also adds a few more array method delegators to the queries.

### CODE REVIEW

File: lib/jobba/status.rb [hunk 1]
```diff
@@ -111,7 +116,8 @@ def set_progress(at, out_of = nil)
     end
 
     def set_job_name(job_name)
-      raise ArgumentError, "`job_name` must not be blank" if job_name.nil? || job_name.empty?
+      raise ArgumentError, "`job_name` must not be blank", caller \
```
Comment (jpslav) [inline lib/jobba/status.rb]: what's the deal with calling `caller` here? (why is this different?)

File: lib/jobba/status.rb [hunk 2]
```diff
@@ -133,6 +140,17 @@ def set_job_args(args_hash={})
       end
     end
 
+    def set_provider_job_id(provider_job_id)
+      raise ArgumentError, "`provider_job_id` must not be blank" \
+        if provider_job_id.nil? || (provider_job_id.respond_to?(:empty?) && provider_job_id.empty?)
```
Comment (jpslav) [inline lib/jobba/status.rb]: and then why not `caller` here?

File: lib/jobba/status.rb (hunk 1 above)
Comment (Dantemss) [inline lib/jobba/status.rb]: When you pass caller to ArgumentError, it's supposed to ommit the `raise` from the backtrace so it starts on whatever called it... I think

File: lib/jobba/status.rb (hunk 2 above)
Comment (Dantemss) [inline lib/jobba/status.rb]: Because I forgot :P

**Comment by Dantemss** [comment]
Added `caller` to all `ArgumentErrors` in jobba.

**Pull request merged**